tempesta_fw/t/functional/helpers/analyzer.py

```diff
@@ -0,0 +1,128 @@
+"""
+Instruments for network traffic analysis
+"""
+from __future__ import print_function
+import os
+from threading import Thread
+from scapy.all import *
+from . import remote, tf_cfg
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+
+FIN = 0x01
+SYN = 0x02
+RST = 0x04
+PSH = 0x08
+ACK = 0x10
+URG = 0x20
+ECE = 0x40
+CWR = 0x80
+
+
+class Sniffer(object):
+
+    def __init__(self, node, host, count=0,
+                 timeout=30, port=80,
+                 node_close=True):
+        self.node = node
+        self.port = port
+        self.thread = None
+        self.captured = 0
+        self.packets = []
+        self.dump_file = '/tmp/tmp_packet_dump'
+        cmd = 'timeout %s tcpdump -i any %s -w - tcp port %s || true'
+        count_flag = ('-c %s' % count) if count else ''
+        self.cmd = cmd % (timeout, count_flag, port)
+        self.err_msg = ' '.join(["Can't %s sniffer on", host])
+        self.node_side_close = node_close
+
+    def sniff(self):
+        '''Thread function for starting system sniffer and saving
+        its output. We need to use temporary file here, because
+        scapy.sniff(offline=file_obj) interface does not support
+        neither StringIO objects nor paramiko file objects.
+        '''
+        stdout, stderr = self.node.run_cmd(self.cmd, timeout=None,
+                                           err_msg=(self.err_msg % 'start'))
+        match = re.search(r'(\d+) packets captured', stderr)
+        if match:
+            self.captured = int(match.group(1))
+        with open(self.dump_file, 'w') as f:
+            f.write(stdout)
+
+    def start(self):
+        self.thread = Thread(target=self.sniff)
+        self.thread.start()
+
+    def stop(self):
+        if self.thread:
+            self.thread.join()
+            if os.path.exists(self.dump_file):
```

Comment:
Missing dump file is a framework error, isn't it? If so we can raise a error here to fail the test and report framework error. See `helpers/error.py`. The test will be marked as `Errored` instead of `Failed`.

Reply:
Corrected.